stable/rocketchat/values.yaml

```diff
@@ -64,6 +64,32 @@ mongodb:
     accessMode: ReadWriteOnce
     size: 8Gi
 
+  replicaSet:
+    ## Whether to create a MongoDB replica set for high availability or not
+    enabled: true
+    useHostnames: true
+  
+    ## Name of the replica set
+    ##
+    name: rs0
```

Comment:
Above you are using rs01 in the connection string.  So either need to change there or here.  I recommend changing here.  Most every where else we use rs01 in our examples